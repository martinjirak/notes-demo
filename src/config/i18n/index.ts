import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import xhr from 'i18next-xhr-backend'
import { initReactI18next } from 'react-i18next'

const options: i18n.InitOptions = {
    load: 'languageOnly',
    ns: ['global'],
    fallbackLng: 'en',
    defaultNS: 'global',
    backend: {
        loadPath: 'public/locales/{{lng}}/{{ns}}.json',
    },
    detection: {
        order: ['localStorage', 'navigator'],
        caches: ['localStorage'],
        lookupLocalStorage: 'i18nextLng',
    },
    interpolation: {
        escapeValue: false,
    },
    react: {
        bindI18n: 'languageChanged',
    },
}

export default i18n.use(LanguageDetector).use(xhr).use(initReactI18next).init(options)
