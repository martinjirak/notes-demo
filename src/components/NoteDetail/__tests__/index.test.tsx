import * as React from 'react'
import { createMount } from '@material-ui/core/test-utils'
import { I18nextProvider } from 'react-i18next'
import { ApolloClient } from 'apollo-client'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { ApolloProvider } from 'react-apollo'
import { RestLink } from 'apollo-link-rest'
import { NoteDetail } from '../'
import i18n from '../../../config/tests/i18n/index'

const restLink = new RestLink({ uri: 'http://private-9aad-note10.apiary-mock.com/' })

const client = new ApolloClient({
    link: restLink,
    cache: new InMemoryCache(),
})

describe('NoteDetail should match Snapshot', () => {
    it('without hint', () => {
        const renderedValue = createMount()(
            <ApolloProvider client={client}>
                <I18nextProvider i18n={i18n}>
                    <NoteDetail />
                </I18nextProvider>
            </ApolloProvider>
        )
        expect(renderedValue.html()).toMatchSnapshot()
    })
})
