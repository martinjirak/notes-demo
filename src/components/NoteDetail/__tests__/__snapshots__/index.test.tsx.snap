// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteDetail should match Snapshot without hint 1`] = `"<div class=\\"makeStyles-root-1\\"><div class=\\"makeStyles-header-2\\"><p class=\\"MuiTypography-root makeStyles-title-3 MuiTypography-body1 MuiTypography-colorPrimary\\"></p></div><p class=\\"MuiTypography-root makeStyles-body-4 MuiTypography-body1\\"></p><div class=\\"makeStyles-buttons-5\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Edit</span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Close</span><span class=\\"MuiTouchRipple-root\\"></span></button></div></div>"`;
